tests/integration/backward_compatible/serialization/compact_compatibility/compact_compatibility_test.py

```diff
@@ -0,0 +1,1691 @@
+
+    def test_set(self):
+        self.map.set(INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE)
+        self.assertEqual(OUTER_COMPACT_INSTANCE, self.map.get(INNER_COMPACT_INSTANCE))
+
+    def test_set_ttl(self):
+        self._put_from_another_client(OUTER_COMPACT_INSTANCE, INNER_COMPACT_INSTANCE)
+        self.map.set_ttl(OUTER_COMPACT_INSTANCE, 999)
+
+    def test_try_lock(self):
+        self.assertTrue(self.map.try_lock(OUTER_COMPACT_INSTANCE))
+
+    def test_try_put(self):
+        self.assertTrue(self.map.try_put(INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE))
+
+    def test_try_remove(self):
+        self.assertFalse(self.map.try_remove(OUTER_COMPACT_INSTANCE))
+        self.map.put(OUTER_COMPACT_INSTANCE, INNER_COMPACT_INSTANCE)
+        self.assertTrue(self.map.try_remove(OUTER_COMPACT_INSTANCE))
+
+    def test_unlock(self):
+        try:
+            self.map.unlock(OUTER_COMPACT_INSTANCE)
+        except IllegalMonitorStateError:
+            # Lock is not locked, but we don't care
+            # about it, as we only want to verify
+            # that we can send the serialized form
+            # of key to the server.
+            pass
+
+    def test_values_with_predicate(self):
+        self.map.put(INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE)
+        self.assertEqual([OUTER_COMPACT_INSTANCE], self.map.values(CompactPredicate()))
+
+    def test_values_with_paging_predicate(self):
+        self.map.put(INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE)
+        self.assertEqual([OUTER_COMPACT_INSTANCE], self.map.values(paging(CompactPredicate(), 1)))
+
+    def _put_from_another_client(self, key, value):
+        other_client = self.create_client(self.client_config)
+        other_client_map = other_client.get_map(self.map.name).blocking()
+        other_client_map.put(key, value)
+
+    def _assert_entry_event(self, events):
+        self._put_from_another_client(INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE)
+
+        def assertion():
+            self.assertEqual(1, len(events))
+            event = events[0]
+            self.assertEqual(INNER_COMPACT_INSTANCE, event.key)
+            self.assertEqual(OUTER_COMPACT_INSTANCE, event.value)
+            self.assertIsNone(event.old_value)
+            self.assertIsNone(event.merging_value)
+
+        self.assertTrueEventually(assertion)
+
+
+class NearCachedMapCompactCompatibilityTest(MapCompatibilityTest):
+    def setUp(self) -> None:
+        map_name = random_string()
+        self.client_config = copy.deepcopy(NearCachedMapCompactCompatibilityTest.client_config)
+        self.client_config["near_caches"] = {map_name: {}}
+        super().setUp()
+        self.map = self.client.get_map(map_name).blocking()
+
+    def test_get_for_near_cache(self):
+        # Another variant of the test in the superclass, where we lookup a key
+        # which has a value whose schema is not fully sent to the
+        # cluster. The near cache will try to serialize it, but it should
```

Comment:
> which has a value whose schema is not fully sent to the cluster.

The value here is the `OUTER_COMPACT_INSTANCE`. What do you mean by schema not fully sent to the cluster? Does not  `_put_from_another_client` send its schema to the cluster ?

Reply:
I meant the schema of the outer compact is not sent to the cluster from this client. let me add this to the comment as well